src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -495,7 +495,7 @@ private function addServiceInlinedDefinitions(string $id, Definition $definition
             // $b = new ServiceB();
             // $a = new ServiceA(ServiceB $b);
             // $b->setServiceA(ServiceA $a);
-            if (isset($inlinedDefinition[$definition]) && $this->hasReference($id, array($def->getArguments(), $def->getFactory()))) {
+            if (isset($inlinedDefinitions[$definition]) && $this->hasReference($id, array($def->getArguments(), $def->getFactory()))) {
```

Comment:
issue doesn't also exist on a lower branch?

Reply:
Yes, lower version branches are not affected.

Reply:
actually, 3.4 is also affected

Reply:
Hm, then I'll create a new PR later today with only this change and think of the second PR for the throwable case.